core/server/master/src/test/java/alluxio/master/file/FileSystemMasterTest.java

```diff
@@ -344,6 +346,43 @@ public void deleteDir() throws Exception {
         mFileSystemMaster.getFileId(new AlluxioURI("/mnt/local/dir1")));
   }
 
+  @Test
+  public void deleteDirRecursiveWithPermissions() throws Exception {
+    // Case 1: userA has permissions to delete directory and file
+    createFileWithSingleBlock(NESTED_FILE_URI);
+    mFileSystemMaster.setAttribute(NESTED_URI,
+        SetAttributeOptions.defaults().setMode((short) 0777));
+    mFileSystemMaster.setAttribute(NESTED_FILE_URI,
+        SetAttributeOptions.defaults().setMode((short) 0777));
+    try (AuthenticatedClientUserResource userA = new AuthenticatedClientUserResource("userA")) {
+      mFileSystemMaster.delete(NESTED_URI, DeleteOptions.defaults().setRecursive(true));
+    }
+    assertEquals(IdUtils.INVALID_FILE_ID, mFileSystemMaster.getFileId(NESTED_URI));
+    assertEquals(IdUtils.INVALID_FILE_ID, mFileSystemMaster.getFileId(NESTED_FILE_URI));
+
+    // Case 2: userA has permissions to delete directory but not one of the files
+    createFileWithSingleBlock(NESTED_FILE_URI);
+    createFileWithSingleBlock(NESTED_FILE2_URI);
+    mFileSystemMaster.setAttribute(NESTED_FILE_URI,
+        SetAttributeOptions.defaults().setMode((short) 0700));
+    mFileSystemMaster.setAttribute(NESTED_FILE2_URI,
+        SetAttributeOptions.defaults().setMode((short) 0777));
+    try (AuthenticatedClientUserResource userA = new AuthenticatedClientUserResource("userA")) {
+      mFileSystemMaster.delete(NESTED_URI, DeleteOptions.defaults().setRecursive(true));
+      Assert.fail("Deleting a directory w/ insufficient permission on child should fail");
```

Comment:
can statically import `fail`, similar to what we do for `assertTrue`, `assertNotEquals, `assertEquals`, etc